### PR TITLE
[PLAT-1841] Regenerate ADDON API TOKEN during reprovisioning

### PR DESCRIPTION
**Pull request by imrekel**
+ added VS Code debugging support

## local:
```
curl -X POST -H "Authentication: addon-access-token" -H "Content-Type: application/json" -d '{"app_slug":"app-slug1","api_token":"bitrise_token1","plan":"free"}' "http://127.0.0.1:5000/provision"
{"envs":[{"key":"ADDON_VDTESTING_API_URL","value":"http://localhost:5000/test"},{"key":"ADDON_VDTESTING_API_TOKEN","value":"gWPn7WGVGLXpDOCsJowGLoirowj2oDzAjRWF1Q27bFbXrlOviW"}]}

curl -X POST -H "Authentication: addon-access-token" -H "Content-Type: application/json" -d '{"app_slug":"app-slug1","api_token":"bitrise_token1","plan":"free"}' "http://127.0.0.1:5000/provision"
{"envs":[{"key":"ADDON_VDTESTING_API_URL","value":"http://localhost:5000/test"},{"key":"ADDON_VDTESTING_API_TOKEN","value":"6lHmXIqCxfLOqMNW7VTvkSU4keJlTt9c8MPBaoTWlgpteJ3dmG"}]}
```

## staging:
```
curl -X POST -H "Authentication: test-token" -H "Content-Type: application/json" -d '{"app_slug":"app-slug1","api_token":"bitrise_token1","plan":"free"}' "https://bitrise-addon-ftl-staging.herokuapp.com/provision"             ✔  
{"envs":[{"key":"ADDON_VDTESTING_API_URL","value":"https://bitrise-addon-ftl-staging.herokuapp.com/test"},{"key":"ADDON_VDTESTING_API_TOKEN","value":"7neXtcHoI1T7oEGnjSFWf4gRvcpjM2QZQCfFTA0mDoljFw1tJo"}]}

curl -X POST -H "Authentication: test-token" -H "Content-Type: application/json" -d '{"app_slug":"app-slug1","api_token":"bitrise_token1","plan":"free"}' "https://bitrise-addon-ftl-staging.herokuapp.com/provision"             ✔  
{"envs":[{"key":"ADDON_VDTESTING_API_URL","value":"https://bitrise-addon-ftl-staging.herokuapp.com/test"},{"key":"ADDON_VDTESTING_API_TOKEN","value":"Ci1okg16LS8XDKs4Dw84E30AORpQS3B8gw9bccra7kRouL7a6m"}]}
```

### CODE REVIEW

File: configs/configs.go [hunk 1]
```diff
@@ -18,8 +18,9 @@ import (
 var (
 	port = ""
 	//
-	shouldSkipBuildAuthorizationWithBitriseAPI = false
-	shouldSkipSessionAuthentication            = false
+	shouldSkipBuildAuthorizationWithBitriseAPI          = false
+	shouldSkipSessionAuthentication                     = false
+	shouldSkipProvisioningBitriseAPIWebhhokRegistration = false
```
Comment (zsolt-marta-bitrise) [inline configs/configs.go:23]:
```suggestion
	shouldSkipProvisioningBitriseAPIWebhookRegistration = false
```

File: actions/provision.go [hunk 2]
```diff
@@ -75,23 +74,35 @@ func ProvisionPostHandler(c buffalo.Context) error {
...
 	if err != nil {
 		logger.Error("Failed to get app from DB", zap.Any("error", errors.WithStack(err)))
 		return c.Render(http.StatusInternalServerError, r.JSON(map[string]string{"error": "Internal error"}))
 	}
 
+	app.APIToken = generateRandomHash(50)
+
+	err = database.UpdateApp(app)
+	if err != nil {
+		logger.Error("Failed to update app in DB", zap.Any("error", errors.WithStack(err)))
```
Comment (zsolt-marta-bitrise) [inline actions/provision.go:102]: AFAIK this should be `logger.Errorw()`

File: README.md [hunk 3]
```diff
@@ -2,6 +2,17 @@
 
 This add-on enables Bitrise.io users to test their apps on Firebase Test Lab using the [Virtual Device Testing for Android step](https://www.bitrise.io/integrations/steps/virtual-device-testing-for-android) in their workflows.
 
+## Development with Visual Studio Code
+
+NOTE: originally, the project was set up with a docker-based development workflow in mind. To support debugging some of the flows using VS Code, a few
+adjustments has been made later. The original development flows still work and are required to fully test the app (see later sections).
```
Comment (zsolt-marta-bitrise) [inline README.md:8]:
```suggestion
NOTE: originally, the project was set up with a docker-based development workflow in mind. To support debugging some of the flows using VS Code, a few
adjustments have been made later. The original development flows still work and are required to fully test the app (see later sections).
```

File: README.md [hunk 4]
```diff
@@ -2,6 +2,17 @@
...
+
+1. clone `example.env` abd rename it to `.env` - this tores the env variables required for running the upp using VS Code
+2. run `docker-compose up -d` to start the local development database
+3. run the upp with the `Debug app` configuration
```
Comment (zsolt-marta-bitrise) [inline README.md:12]:
```suggestion
3. run the app with the `Debug app` configuration
```

Comment (imrekel) [reply README.md:12]: you are humiliating me :D

**Review by zsolt-marta-bitrise — APPROVED** [review]

**Pull request merged**